### PR TITLE
Add option to raise or return errors

### PR DESCRIPTION
**Pull request by kookster**
Will raise them buy default, this is an alternate way of implementing #6 by @alexdean solving issue #5

### CODE REVIEW

**Comment by kookster** [comment]
(just fixed the request options handling, and added a test)

So this should work now too since you can pass in `options` on any request, with this as a supported option you can now use it on any request:
```
  api = Amara::API.new(raise_errors: true)
  api.response = api.list(limit: 2, options: { raise_errors: false } )
```
should override the api option

**Comment by kookster** [comment]
as a compromise, I'll try adding the bang methods to the other PR #8

**Comment by alexdean** [comment]
So if i understand correctly, we're doing all the things!

  1. `get`/`get!` (and friends) are all available
  1. behavior of non-bang methods (`get` & etc) can be configured either via constructor or at runtime
  1. raw responses from Amara's api are added to raised `Amara::Error` instances.

I don't see tests for that third one. Otherwise, sure I'm good with this. I don't think doing all of these are really needed - I can only see using a subset, as discussed at length in #6. But that's no dealbreaker and to avoid a bikeshed I say 🚢 it.

@kookster Thank you!

File: test/client_test.rb [hunk 1]
```diff
@@ -37,32 +37,40 @@
...
 
   it "throws an unknown error on all other errors" do
     stub_request(:get, 'https://www.amara.org/api2/partners/api/?limit=2&offset=0').
        to_return(body: nil, status: 601)
-    proc {
+    error = proc {
       Amara::API.new.list(limit: 2)
     }.must_raise Amara::UnknownError
+
+    error.response.status.must_equal 601
```
Comment (kookster) [inline test/client_test.rb]: example of testing the error includes a response.

**Comment by kookster** [comment]
the reason for doing "all the things", in my mind, is to avoid breaking the way it currently works, while giving people the option to use it in a different manner. I too could have done w/o all of these, but I think the ones I could do w/o may not match the ones you would do w/o, so compromise!

**Pull request merged**